WordPress/gutenberg
# [RNMobile] Use ActivityIndicator to display media upload progress

Addresses:

* https://github.com/wordpress-mobile/gutenberg-mobile/issues/6603

## What?
Updates the mobile MediaProgressUpload component to use a spinner in place of a progress bar. Also includes a checkmark success state.


## Why?
As part of ongoing efforts to improve the media upload experience, we have received feedback that Image upload process could be improved by modifying an "in progress" state when uploading Images.

This PR focuses on UI display only, and intentionally does not update any of the media progress upload logic in order to reduce regressions.


## How?
- Removes opacity changes when the Image is being uploaded
- Replaces the horizontal progress bar with an ActivityIndicator spinner in MediaUploadProgress

## Testing Instructions

1. Create a post or page and insert an **Image** or **Gallery** block
2. Add an image from the device
3. Observe that the progress bar is not displayed.
4. Observe that a spinner is displayed in the bottom right corner of the upload.


### Testing Instructions for Keyboard


## Screenshots or screencast 

<img src="https://github.com/WordPress/gutenberg/assets/643285/c43719ea-056a-49c0-bb89-199d14e88e66" />

## Todo

Currently this change applies to all media blocks:

- [ ] Update logic so this change only applies to Image and Gallery blocks
- [ ] Update E2E tests
- [ ] Check snapshot tests needing updates
